x-pack/test_serverless/api_integration/test_suites/observability/slos/create_slo.ts

```diff
@@ -0,0 +1,325 @@
+              total: 'system.cpu.total.norm.pct: *',
+              timestampField: '@timestamp',
+            },
+          },
+          timeWindow: {
+            duration: '7d',
+            type: 'rolling',
+          },
+          budgetingMethod: 'occurrences',
+          objective: {
+            target: 0.999,
+          },
+          groupBy: '*',
+        });
+
+        const createdSlo = await sloApi.waitForSloCreated({ sloId });
+        expect(createdSlo.id).to.be(sloId);
+      });
+    });
+
+    describe('SLO with special characters in the description', () => {
+      it("creates an SLO that has ' character in the description", async () => {
+        const sloId = 'my-custom-id3';
+        await sloApi.create({
+          id: sloId,
+          name: 'my SLO with weird characters in the description',
+          description:
+            "Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged.",
+          indicator: {
+            type: 'sli.kql.custom',
+            params: {
+              index: infraDataIndex,
+              good: 'system.cpu.total.norm.pct > 1',
+              total: 'system.cpu.total.norm.pct: *',
+              timestampField: '@timestamp',
+            },
+          },
+          timeWindow: {
+            duration: '7d',
+            type: 'rolling',
+          },
+          budgetingMethod: 'occurrences',
+          objective: {
+            target: 0.999,
+          },
+          groupBy: '*',
+        });
+        const createdSlo = await sloApi.waitForSloCreated({ sloId });
+        expect(createdSlo.id).to.be(sloId);
+      });
+    });
+
+    describe('partition by SLO', () => {
+      it('creates a partition by SLO', async () => {
+        const sloId = 'my-custom-id4';
+        await sloApi.create({
+          id: sloId,
+          name: 'Group by SLO',
+          description: 'This is a group by SLO.',
+          indicator: {
+            type: 'sli.kql.custom',
+            params: {
+              index: infraDataIndex,
+              good: 'system.cpu.total.norm.pct > 1',
+              total: 'system.cpu.total.norm.pct: *',
+              timestampField: '@timestamp',
+            },
+          },
+          timeWindow: {
+            duration: '7d',
+            type: 'rolling',
+          },
+          budgetingMethod: 'occurrences',
+          objective: {
+            target: 0.999,
+          },
+          groupBy: 'host.name',
+        });
+        const createdSlo = await sloApi.waitForSloCreated({ sloId });
+        expect(createdSlo.id).to.be(sloId);
+        expect(createdSlo.groupBy).not.to.be(ALL_VALUE);
+        expect(createdSlo.groupBy).to.be('host.name');
+      });
+    });
+
+    describe('Total transforms', () => {
```

Comment:
Idea was to verify total number of transforms and SO definitions at the end of all tests. Not sure if that's a good idea, since if I run these tests isolated (`it.only`) they will of course fail. I would like some feedback on following:

1. Is it ok to leave them, even if they fail, in case they run isolated? Is there like a rule/best practise to always have tests be green when run isolated?
2. Or shall I better add these expectations in each test separately and remove them from the end?

Reply:
I feel like each test should run in isolation unless it's some kind of multi-step test that relies on the previous results, then clean up at the suite level. For these tests, I'd probably clean up the transforms after each test just to make sure we leave the test environment in a clean state.